python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -331,22 +309,25 @@ def __getitem__(self, this_id):
             this_id (int): This is the key of an object in the object dictionary
         """
         logger.debug("__getitem__")
+        # Is this an Incomplete Object?
+        if this_id in self._incomplete_objects:
+            return IncompleteObject(self, this_id)
+
+        # Nope, so try to find it and raise an exception if not
         try:
             return self._objects[this_id]
         except KeyError as err:
             logger.debug("Repo KeyError: %s" % err)
             logger.debug("Keys: %s id: %s" % (self._objects.keys(), this_id))
-            if this_id in self._incomplete_objects:
```

Comment:
I'm referring to the function as a whole as in the new version, I've just moved the `incomplete_objects` check outside of the `KeyError` to before the `try..except` block so this check occurs **before** the check against `_objects`. This function will only give different behaviour if there is a point where the object is referenced in both lists as it will return an `IncompleteObject` rather than the copy in `objects`.